subprojects/signing/src/main/java/org/gradle/plugins/signing/signatory/pgp/PgpSignatoryFactory.java

```diff
@@ -81,8 +81,16 @@ protected PgpSignatory readProperties(Project project, String prefix, String nam
         for (String property : PROPERTIES) {
             String qualifiedProperty = (String)getQualifiedPropertyName(prefix, property);
             if (project.hasProperty(qualifiedProperty)) {
-                values.add(
-                    project.property(qualifiedProperty));
+                Object prop = project.property(qualifiedProperty);
+                if(prop != null){
+                    values.add(prop);
+                } else {
+                    if(required){
```

Comment:
I made an attempt at refactoring in 1019328105. It reads clearer but exposes some sentiment I captured in the commit message: this capability to retrieve a property safely probably belongs to Project as an overloaded method.